google/cloud/sql/connector/connector.py

```diff
@@ -137,17 +166,21 @@ def connect(
             )
         else:
             ip_type = kwargs.pop("ip_type", self._ip_type)
-        if "timeout" in kwargs:
-            return icm.connect(driver, ip_type, **kwargs)
-        elif "connect_timeout" in kwargs:
+        timeout = kwargs.pop("timeout", self._timeout)
+        if "connect_timeout" in kwargs:
             timeout = kwargs["connect_timeout"]
-        else:
-            timeout = self._timeout
+
         try:
-            return icm.connect(driver, ip_type, timeout, **kwargs)
+            connection_task = self._loop.create_task(
+                icm.connect(driver, ip_type, **kwargs)
+            )
+            await asyncio.wait_for(connection_task, timeout)
+            return await connection_task
+        except asyncio.TimeoutError:
+            raise TimeoutError(f"Connection timed out after {timeout}s")
         except Exception as e:
             # with any other exception, we attempt a force refresh, then throw the error
-            icm.force_refresh()
+            await self._loop.create_task(icm.force_refresh())
```

Comment:
@kurtisvg should this just be `await icm.force_refresh()`? There is no benefit of creating a task to await it immediately right? Just double checking.